esp-hal-procmacros/src/blocking_main.rs

```diff
@@ -0,0 +1,31 @@
+#![deny(warnings)]
+
+use proc_macro::TokenStream;
+use proc_macro2::Span;
+use syn::{
+    parse::{self},
+    parse_macro_input,
+    ItemFn,
+};
+
+pub fn main(args: TokenStream, input: TokenStream) -> TokenStream {
+    let f = parse_macro_input!(input as ItemFn);
+
+    if f.sig.asyncness.is_some() {
+        return parse::Error::new(Span::call_site(), "If you want to use `async` please use `esp-hal-embassy` (https://docs.rs/esp-hal-embassy/latest/esp_hal_embassy/)")
```

Comment:
Could we expand this a bit to also give a direct code snippet, something like `#[esp_hal_embassy::main]`